src/frontEnd/volunteering/ViewVolunteeringOption.tsx

```diff
@@ -0,0 +1,24 @@
+import React from 'react';
+import { Page } from '../page/Page';
+import  { LoadVolunteeringOption } from "./LoadVolunteeringOption";
+import { List } from "../../utils/list/List";
+
+
+export const ViewVolunteeringOption = ({url, organizations, navTo}) => {
+    const selectedJobID = (url.path).substring(14);
+// alternative of filtering data again : if we can upsate state to have selected data when 
+// onSuggestionSelected is called, so that we can directly use it to load next page.
+    const selectedData = organizations.map(org => {
```

Comment:
Ok let me see, so the idea here is that given all organizations, get the org and vol opportunity matching the id

list(organizations).flatMap(o => o.volOpportunities.map(v => ({v,o}))).first(({v}) => v.id == selectedJobId)

This will give you a maybe that, if found, is the org and volunteering opportunity pair you are looking for.